validation/views.py

```diff
@@ -210,58 +200,42 @@ def advanced_search_results(request):
+    filter_query = []
+    if transcription:
+        filter_query.append(Q(transcription__contains=transcription))
+    if translation:
+        filter_query.append(Q(translation__contains=translation))
+    if analysis:
+        filter_query.append(Q(analysis__contains=analysis))
+    if status and status != "all":
+        filter_query.append(Q(status=status))
+
+    if filter_query:
+        phrase_matches = Phrase.objects.filter(
+            reduce(operator.or_, filter_query)
         ).prefetch_related("recording_set__speaker")
     else:
-        cree_matches = []
-
-    if translation != "":
-        english_matches = Phrase.objects.filter(
-            translation__contains=translation
-        ).prefetch_related("recording_set__speaker")
-    else:
-        english_matches = []
-
-    if analysis != "":
-        analysis_matches = Phrase.objects.filter(
-            analysis__contains=analysis
-        ).prefetch_related("recording_set__speaker")
-    else:
-        analysis_matches = []
-
-    if transcription == "" and translation == "" and analysis == "":
         phrase_matches = Phrase.objects.all().prefetch_related("recording_set__speaker")
-    else:
-        phrase_matches = list(
-            set().union(cree_matches, english_matches, analysis_matches)
-        )
-
-    if status != "all":
-        if status == "new":
-            status_matches = Phrase.objects.filter(status="new")
-        elif status == "linked":
-            status_matches = Phrase.objects.filter(status="linked")
-        elif status == "auto-val":
-            status_matches = Phrase.objects.filter(status="auto-validated")
-        phrase_and_status_matches = list(
-            set(phrase_matches).intersection(status_matches)
-        )
-    else:
-        phrase_and_status_matches = phrase_matches
 
     recordings = {}
     all_matches = []
-    for phrase in phrase_and_status_matches:
-        recordings[phrase] = []
-        if ("all" in speakers or speakers == []) and (quality == "all" or not quality):
-            recordings[phrase] = list(phrase.recordings)
-        else:
-            for recording in phrase.recordings:
-                if recording.speaker.code in speakers or recording.quality == quality:
-                    recordings[phrase].append(recording)
 
+    # We use the negation of the query, ~Q, and .exclude here
+    # because we want to remove any entries and recordings that do not match
+    # the criteria, which cannot be accomplished with .filter
+    recordings_filter_query = []
+    if speakers and "all" not in speakers:
+        recordings_filter_query.append(~Q(speaker__in=speakers))
+    if quality and "all" not in quality:
+        recordings_filter_query.append(~Q(quality__in=quality))
+
+    for phrase in phrase_matches:
+        if recordings_filter_query:
+            recordings[phrase] = phrase.recordings.exclude(
+                reduce(operator.and_, recordings_filter_query)
+            )
```

Comment:
Since this is an `exclude` query, not a `filter` one, I would like to suggest that the name should match:

```suggestion
    recordings_exclude_query = []
    if speakers and "all" not in speakers:
        recordings_exclude_query.append(~Q(speaker__in=speakers))
    if quality and "all" not in quality:
        recordings_exclude_query.append(~Q(quality__in=quality))

    for phrase in phrase_matches:
        if recordings_exclude_query:
            recordings[phrase] = phrase.recordings.exclude(
                reduce(operator.and_, recordings_exclude_query)
            )
```

Reply:
Sure thing!